rasa/nlu/featurizers/dense_featurizer/lm_featurizer.py

```diff
@@ -754,15 +753,20 @@ def train(
                     self._set_lm_features(batch_docs[index], ex, attribute)
                 batch_start_index += batch_size
 
-    def process(self, message: Message, **kwargs: Any) -> None:
-        """Process an incoming message by computing its tokens and dense features.
+        return training_data
 
-        Args:
-            message: Incoming message object
-        """
-        # process of all featurizers operates only on TEXT and ACTION_TEXT attributes,
-        # because all other attributes are labels which are featurized during training
-        # and their features are stored by the model itself.
+    def process(self, messages: List[Message]) -> List[Message]:
+        """Processes messages by computing tokens and dense features."""
+        for message in messages:
+            if message:
```

Comment:
You can remove the "if" here (because messages won't evaluate to False and) because the handling of empty lists is achieved by the for loop already:  " for message in []: # do xy " won't ever execute xy :)

Reply:
Maybe I misunderstood: in the requirements we say to handle empty messages (#9456).
I though this meant that some messages will be `None`. In this case, the loop will not skip them but the `if` statement will. Or is that not the required behaviour? 🤔

Reply:
I mean not the whole list is empty, but some individual messages are empty.

Reply:
Messages won't be empty 👍🏻 With "empty messages" we meant empty lists. Sorry for the vagueness!

Reply:
Oh, no problem!